src/string/utf16.cr

```diff
@@ -12,22 +12,30 @@ class String
   # "hi 𐂥".to_utf16 # => Slice[104_u16, 105_u16, 32_u16, 55296_u16, 56485_u16]
   # ```
   def to_utf16 : Slice(UInt16)
-    size = 0
-    each_char do |char|
-      size += char.ord < 0x10000 ? 1 : 2
+    if ascii_only?
+      # size == bytesize, so each char fits in one UInt16
+      u16_size = bytesize
```

Comment:
If it's ascii only, could we maybe do `to_slice.map &.ord.to_u16` or something like that? I don't know if that's faster, maybe yes (no need to compare values in the loop) but then we can bail out earlier.

Reply:
Yep, that's even better 👍